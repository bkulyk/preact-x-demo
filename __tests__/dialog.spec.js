import React from 'react';
import Dialog from '@bitchin/dialog';
import { shallow } from './enzyme';
import { expect } from './jest';

it('renders correctly with defaults', () => {
  const tree = shallow((
    <Dialog
      title="test dialog"
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with component actions', () => {
  const tree = shallow((
    <Dialog
      title="test dialog"
      actions={(
        <div>actions would go here</div>
      )}
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with array actions', () => {
  const tree = shallow((
    <Dialog
      title="test dialog"
      actions={['cancel', 'ok']}
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with hashmap actions', () => {
  const tree = shallow((
    <Dialog
      title="test dialog"
      actions={{ cancel: 'Cancel', ok: 'OK' }}
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with hashmap actions and a default action', () => {
  const tree = shallow((
    <Dialog
      title="test dialog"
      actions={{ cancel: 'Cancel', ok: 'OK' }}
      defaultAction="ok"
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with a simple body', () => {
  const tree = shallow((
    <Dialog
      title="simple body"
      body="lorem ipsum dolor sit"
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with a custom body', () => {
  const tree = shallow((
    <Dialog
      title="simple body"
      body={(<em>lorem ipsum dolor sit</em>)}
    />
  ));

  expect(tree).toMatchSnapshot();
});

it('renders correctly with a custom title', () => {
  const tree = shallow((
    <Dialog
      title={(<strong>strong body</strong>)}
    />
  ));

  expect(tree).toMatchSnapshot();
});
