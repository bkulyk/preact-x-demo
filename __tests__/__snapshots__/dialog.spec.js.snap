// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a custom body 1`] = `
<div
  aria-describedby="my-dialog-content"
  aria-labelledby="my-dialog-title"
  aria-modal="true"
  className="mdc-dialog"
  role="alertdialog"
>
  <div
    className="mdc-dialog__container"
  >
    <div
      className="mdc-dialog__surface"
    >
      <h2
        className="mdc-dialog__title"
        id="my-dialog-title"
      >
        simple body
      </h2>
      <div
        className="mdc-dialog__content"
        id="my-dialog-content"
      >
        <em>
          lorem ipsum dolor sit
        </em>
      </div>
      <footer
        className="mdc-dialog__actions"
      >
        <DialogActions
          actions={
            Object {
              "ok": "OK",
            }
          }
          defaultAction={null}
        />
      </footer>
    </div>
  </div>
  <div
    className="mdc-dialog__scrim"
  />
</div>
`;

exports[`renders correctly with a custom title 1`] = `
<div
  aria-describedby="my-dialog-content"
  aria-labelledby="my-dialog-title"
  aria-modal="true"
  className="mdc-dialog"
  role="alertdialog"
>
  <div
    className="mdc-dialog__container"
  >
    <div
      className="mdc-dialog__surface"
    >
      <h2
        className="mdc-dialog__title"
        id="my-dialog-title"
      >
        <strong>
          strong body
        </strong>
      </h2>
      <div
        className="mdc-dialog__content"
        id="my-dialog-content"
      />
      <footer
        className="mdc-dialog__actions"
      >
        <DialogActions
          actions={
            Object {
              "ok": "OK",
            }
          }
          defaultAction={null}
        />
      </footer>
    </div>
  </div>
  <div
    className="mdc-dialog__scrim"
  />
</div>
`;

exports[`renders correctly with a simple body 1`] = `
<div
  aria-describedby="my-dialog-content"
  aria-labelledby="my-dialog-title"
  aria-modal="true"
  className="mdc-dialog"
  role="alertdialog"
>
  <div
    className="mdc-dialog__container"
  >
    <div
      className="mdc-dialog__surface"
    >
      <h2
        className="mdc-dialog__title"
        id="my-dialog-title"
      >
        simple body
      </h2>
      <div
        className="mdc-dialog__content"
        id="my-dialog-content"
      >
        lorem ipsum dolor sit
      </div>
      <footer
        className="mdc-dialog__actions"
      >
        <DialogActions
          actions={
            Object {
              "ok": "OK",
            }
          }
          defaultAction={null}
        />
      </footer>
    </div>
  </div>
  <div
    className="mdc-dialog__scrim"
  />
</div>
`;

exports[`renders correctly with array actions 1`] = `
<div
  aria-describedby="my-dialog-content"
  aria-labelledby="my-dialog-title"
  aria-modal="true"
  className="mdc-dialog"
  role="alertdialog"
>
  <div
    className="mdc-dialog__container"
  >
    <div
      className="mdc-dialog__surface"
    >
      <h2
        className="mdc-dialog__title"
        id="my-dialog-title"
      >
        test dialog
      </h2>
      <div
        className="mdc-dialog__content"
        id="my-dialog-content"
      />
      <footer
        className="mdc-dialog__actions"
      >
        <DialogActions
          actions={
            Array [
              "cancel",
              "ok",
            ]
          }
          defaultAction={null}
        />
      </footer>
    </div>
  </div>
  <div
    className="mdc-dialog__scrim"
  />
</div>
`;

exports[`renders correctly with component actions 1`] = `
<div
  aria-describedby="my-dialog-content"
  aria-labelledby="my-dialog-title"
  aria-modal="true"
  className="mdc-dialog"
  role="alertdialog"
>
  <div
    className="mdc-dialog__container"
  >
    <div
      className="mdc-dialog__surface"
    >
      <h2
        className="mdc-dialog__title"
        id="my-dialog-title"
      >
        test dialog
      </h2>
      <div
        className="mdc-dialog__content"
        id="my-dialog-content"
      />
      <footer
        className="mdc-dialog__actions"
      >
        <div>
          actions would go here
        </div>
      </footer>
    </div>
  </div>
  <div
    className="mdc-dialog__scrim"
  />
</div>
`;

exports[`renders correctly with defaults 1`] = `
<div
  aria-describedby="my-dialog-content"
  aria-labelledby="my-dialog-title"
  aria-modal="true"
  className="mdc-dialog"
  role="alertdialog"
>
  <div
    className="mdc-dialog__container"
  >
    <div
      className="mdc-dialog__surface"
    >
      <h2
        className="mdc-dialog__title"
        id="my-dialog-title"
      >
        test dialog
      </h2>
      <div
        className="mdc-dialog__content"
        id="my-dialog-content"
      />
      <footer
        className="mdc-dialog__actions"
      >
        <DialogActions
          actions={
            Object {
              "ok": "OK",
            }
          }
          defaultAction={null}
        />
      </footer>
    </div>
  </div>
  <div
    className="mdc-dialog__scrim"
  />
</div>
`;

exports[`renders correctly with hashmap actions 1`] = `
<div
  aria-describedby="my-dialog-content"
  aria-labelledby="my-dialog-title"
  aria-modal="true"
  className="mdc-dialog"
  role="alertdialog"
>
  <div
    className="mdc-dialog__container"
  >
    <div
      className="mdc-dialog__surface"
    >
      <h2
        className="mdc-dialog__title"
        id="my-dialog-title"
      >
        test dialog
      </h2>
      <div
        className="mdc-dialog__content"
        id="my-dialog-content"
      />
      <footer
        className="mdc-dialog__actions"
      >
        <DialogActions
          actions={
            Object {
              "cancel": "Cancel",
              "ok": "OK",
            }
          }
          defaultAction={null}
        />
      </footer>
    </div>
  </div>
  <div
    className="mdc-dialog__scrim"
  />
</div>
`;

exports[`renders correctly with hashmap actions and a default action 1`] = `
<div
  aria-describedby="my-dialog-content"
  aria-labelledby="my-dialog-title"
  aria-modal="true"
  className="mdc-dialog"
  role="alertdialog"
>
  <div
    className="mdc-dialog__container"
  >
    <div
      className="mdc-dialog__surface"
    >
      <h2
        className="mdc-dialog__title"
        id="my-dialog-title"
      >
        test dialog
      </h2>
      <div
        className="mdc-dialog__content"
        id="my-dialog-content"
      />
      <footer
        className="mdc-dialog__actions"
      >
        <DialogActions
          actions={
            Object {
              "cancel": "Cancel",
              "ok": "OK",
            }
          }
          defaultAction="ok"
        />
      </footer>
    </div>
  </div>
  <div
    className="mdc-dialog__scrim"
  />
</div>
`;
